pkg/operator/staticresource.go

```diff
@@ -0,0 +1,171 @@
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+// CSIStaticResourceController creates, manages and deletes static resources of a CSI driver, such as RBAC rules.
+// It's more hardcoded variant of library-go's StaticResourceController, which does not implement removal
+// of objects yet.
+type CSIStaticResourceController struct {
+	operatorName      string
+	operatorNamespace string
+	operatorClient    operatorv1helpers.OperatorClientWithFinalizers
+	kubeClient        kubernetes.Interface
+	eventRecorder     events.Recorder
+
+	// Objects to sync
+	csiDriver          *storagev1.CSIDriver
+	nodeServiceAccount *corev1.ServiceAccount
+	nodeRole           *rbacv1.ClusterRole
+	nodeRoleBinding    *rbacv1.ClusterRoleBinding
+}
+
+func NewCSIStaticResourceController(
+	name string,
+	operatorNamespace string,
+	operatorClient operatorv1helpers.OperatorClientWithFinalizers,
+	kubeClient kubernetes.Interface,
+	informers operatorv1helpers.KubeInformersForNamespaces,
+	recorder events.Recorder,
+	csiDriver *storagev1.CSIDriver,
+	nodeServiceAccount *corev1.ServiceAccount,
+	nodeRole *rbacv1.ClusterRole,
+	nodeRoleBinding *rbacv1.ClusterRoleBinding,
+) factory.Controller {
+	c := &CSIStaticResourceController{
+		operatorName:       name,
+		operatorNamespace:  operatorNamespace,
+		operatorClient:     operatorClient,
+		kubeClient:         kubeClient,
+		eventRecorder:      recorder,
+		csiDriver:          csiDriver,
+		nodeServiceAccount: nodeServiceAccount,
+		nodeRole:           nodeRole,
+		nodeRoleBinding:    nodeRoleBinding,
+	}
+
+	operatorInformers := []factory.Informer{
+		operatorClient.Informer(),
+		informers.InformersFor(operatorNamespace).Core().V1().ServiceAccounts().Informer(),
+		informers.InformersFor(operatorNamespace).Storage().V1().CSIDrivers().Informer(),
+		informers.InformersFor(operatorNamespace).Rbac().V1().ClusterRoles().Informer(),
+		informers.InformersFor(operatorNamespace).Rbac().V1().ClusterRoleBindings().Informer(),
+	}
+	return factory.New().
+		WithSyncDegradedOnError(operatorClient).
+		WithInformers(operatorInformers...).
+		WithSync(c.sync).
+		ResyncEvery(time.Minute).
+		ToController(name, recorder.WithComponentSuffix("csi-static-resource-controller"))
+}
+
+func (c *CSIStaticResourceController) sync(ctx context.Context, controllerContext factory.SyncContext) error {
+	opSpec, opStatus, _, err := c.operatorClient.GetOperatorState()
+	if apierrors.IsNotFound(err) {
+		// TODO: proceed with removal?
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if opSpec.ManagementState != opv1.Managed {
+		return nil
+	}
+
+	meta, err := c.operatorClient.GetObjectMeta()
+	if err != nil {
+		return err
+	}
+	if meta.DeletionTimestamp != nil {
```

Comment:
Why in [other controllers](https://github.com/bertinatto/library-go/blob/bd06a0ee02c366b7a0578e5811147b2252676568/pkg/operator/csi/csidrivernodeservicecontroller/csi_driver_node_service_controller.go#L136) we do `IsOperatorRemovable()` and here we don't?

Reply:
because the whole point of this controller is to be removable. But I fixed it.